modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/InputParametersTypeModelValidator.java

```diff
@@ -0,0 +1,141 @@
+import static org.mule.runtime.extension.api.util.NameUtils.getComponentModelTypeName;
+import static org.mule.runtime.module.extension.internal.util.IntrospectionUtils.getFieldsWithGetters;
+import org.mule.metadata.api.model.ArrayType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.api.model.ObjectFieldType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.NamedObject;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.api.meta.model.source.SourceCallbackModel;
+import org.mule.runtime.api.meta.model.source.SourceModel;
+import org.mule.runtime.api.meta.model.util.IdempotentExtensionWalker;
+import org.mule.runtime.extension.api.declaration.type.TypeUtils;
+import org.mule.runtime.extension.api.loader.ExtensionModelValidator;
+import org.mule.runtime.extension.api.loader.Problem;
+import org.mule.runtime.extension.api.loader.ProblemsReporter;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Validates the types used as:
+ *
+ * <ul>
+ * <li>Parameter of an {@link OperationModel}</li>
+ * <li>Parameter of an {@link SourceCallbackModel}</li>
+ * </ul>
+ *
+ * These types should comply that for each parameter they contain, there must exist a getter method for that parameter.
+ * The vice versa is also required, that implies the fact that for each getter method, there should exist a field named with the
+ * getter convention.
+ *
+ * @since 4.0
+ */
+public final class InputParametersTypeModelValidator implements ExtensionModelValidator {
+
+  @Override
+  public void validate(ExtensionModel extensionModel, ProblemsReporter problems) {
+    final Set<Class<?>> validatedTypes = new HashSet<>();
+    new IdempotentExtensionWalker() {
+
+      @Override
+      public void onOperation(OperationModel model) {
+        model.getAllParameterModels()
+            .forEach(parameterModel -> validateJavaType(model, parameterModel.getType(), problems, validatedTypes));
+      }
+
+      @Override
+      public void onSource(SourceModel model) {
+        validateCallback(model, model.getSuccessCallback());
+        validateCallback(model, model.getErrorCallback());
+        validateCallback(model, model.getTerminateCallback());
+      }
+
+      private void validateCallback(SourceModel model, Optional<SourceCallbackModel> callback) {
+        callback.ifPresent(cb -> cb.getAllParameterModels()
+            .forEach(parameterModel -> validateJavaType(model, parameterModel.getType(), problems, validatedTypes)));
+      }
+    }.walk(extensionModel);
+
+    extensionModel.getSubTypes().forEach(subTypesModel -> getClassForValidation(subTypesModel.getBaseType())
+        .filter(validatedTypes::contains).ifPresent(type -> {
```

Comment:
this will ONLY validate the oneas already validated. You seem to be missing a not and to not have good enough tests